agent/command/generate.go

```diff
@@ -114,13 +114,20 @@ func (c *generateTask) Execute(ctx context.Context, comm client.Communicator, lo
 			if err != nil {
 				return true, err
 			}
+
+			var generateErr error
 			if len(generateStatus.Errors) > 0 {
-				fullErr := errors.New(strings.Join(generateStatus.Errors, ", "))
+				generateErr = errors.New(strings.Join(generateStatus.Errors, ", "))
+			}
+			if !generateStatus.ShouldRetry {
```

Comment:
I think this relies on a perfectly lockstep deploy. Consider:
1. We deploy. The app servers and agents both (eventually) get the new code.
2. We revert. The app servers revert, but not yet all agents.
In that state `ShouldRetry` will always be false. Agents that have this code will always stop at 123 and never get to 125.

Reply:
I see what you're saying. I'll flip this boolean so that we retry with the default value.